### DISCUSSION TITLE
How to specify payload size

### DISCUSSION THREAD

**Discussion by leisun124** [eProsima/Fast-DDS#4110] [General]
with data topic idl like this:
struct myData
{
   unsigned long len;
   octect message[1024];
}
The message size can be really small. say 10 bytes, but fast-dds always serialize the entire 1024 bytes to network, isn't it?  I can't use sequence octet, as that translated to std::vector<uint8_t> , and I can't use zerocopy since the data type is not plain anymore.
Any idea to improve to get best performance? 
THank
Lei

**Comment by JesusPoderoso** [comment]
Hi @leisun124, thanks for using Fast DDS.
You need to use [bounded types among other constraints](https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/datasharing.html#constraints) if you want to use the Zero-Copy mechanism with Data-sharing.
If your deployment is in the same host, you can consider using Zero-Copy mechanism with [Intra-process delivery](https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/intraprocess.html#intra-process-delivery). 
If that does not suit you, your remaining option is Shared Memory Transport.
You can check in [this discussion thread](https://github.com/eProsima/Fast-DDS/discussions/3654#discussioncomment-6372721) the benefits / amount of data copies per delivery method.
